src/components/cardRegister/CardExpireDateInput.tsx

```diff
@@ -0,0 +1,150 @@
+import React, { useEffect } from "react";
+
+import { currentDate } from "utils/currentDate";
+import { RULE_INPUT } from "constants/constants";
+import {
+  InputBasic,
+  InputBox,
+  InputTitle,
+  InputContainer,
+} from "components/common";
+import { expireDate } from "types";
+
+interface CardExpireDateInputComponent {
+  expireDate: expireDate;
+  isValid: boolean;
+  handleExpireDateInput: (
+    state: expireDate | ((prevState: expireDate) => expireDate)
+  ) => void;
+  handleCardExpireCheck: (isValid: boolean) => void;
+}
+
+export const CardExpireDateInput = ({
+  expireDate,
+  isValid,
+  handleExpireDateInput,
+  handleCardExpireCheck,
+}: CardExpireDateInputComponent) => {
+  useEffect(() => {
+    if (expireDate.year === "" || expireDate.month === "") {
+      return;
+    }
+
+    const isValidate = isCardExpireDateValidate(expireDate);
+    handleCardExpireCheck(isValidate);
+  }, [expireDate]);
+
+  const handleMonthInput = (e: React.ChangeEvent<HTMLInputElement>) => {
+    const { value } = e.target;
+
+    if (Number.isNaN(value) || parseInt(value) > 12) {
+      return;
+    }
```

Comment:
저의 아주 개인적인 의견이지만,  
`parseInt(value)>12 `는 validation 의 영역이고,
`Number.isNaN(value)` 은 타입가드의 영역인 것 같은데, 나눠서 처리하는게 어떨까요?

그리고, value 가 string 타입인 것 같은데 이 경우 어떤 문자열을 넣어도 `Number.isNaN` 의 결과값이 false 로 나오지 않나용..?!

```suggestion

    const valueAsNumber = Number(value);
    
    if(Number.isNaN(valueAsNumber) return;
    if(valueAsNumber > 12) return;
   
```

Reply:
`Number`를 앞에 붙여주다 보니 형변환이 먼저 일어날거라고 생각했었나봐요... 멍청했습니다

타입가드와 validation을 나눠주는 아이디어도 좋은 것 같아요!

처음엔 input type을 number로 받아 형변환 자체를 없애주는 방법도 생각했었는데
type을 text로 두고, pattern을 통해 검사하는게 사용성 측면에서 더 좋게 느껴져서 포기했었던 기억이나네요.
결론은 계속 고민하다가... 안 고치고 넘어간 것 같습니다..^^ 😅

- [refactor: 타입가드와 validation을 분리](https://github.com/usageness/react-payments/pull/1/commits/5b7f46606420e0485dac8014c3cb6ce8ef02a971)

Reply:
셀프회고 좋습니다. 좋은 리뷰이의 자세네요. 합격입니다.